### PR TITLE
add bookmarks to bc and pf scrolls

### LINKED ISSUE #205: Add bookmarks to pointfog

### LINKED ISSUE #206: Add bookmarks to scroll bar

### PR DESCRIPTION
**Pull request by j6k4m8**
fixes #206 and fixes #205. 

image

### CODE REVIEW

**Comment by j6k4m8** [comment]
Also note the thumb now sticks out a bit to the left, and bookmarks stick out a bit to the right, to prevent them from getting buried in denser volumes.

File: breadcrumbs/src/layers/Scrollbar.js [hunk 1]
```diff
@@ -36,24 +37,42 @@ export default class Scrollbar {
         this.width = 20;
     }
 
-    mouseClicked(): void {}
-    mousePressed(): void {}
+    mouseClicked(): boolean {
+
+    }
+    mousePressed(): void {
+        console.log("it happened.");
```
Comment (cabishop) [inline breadcrumbs/src/layers/Scrollbar.js]: @j6k4m8 - Debug print statement can be removed

File: breadcrumbs/src/layers/Scrollbar.js [hunk 2]
```diff
@@ -36,24 +37,42 @@ export default class Scrollbar {
...
                 z: i.z,
-                color: c
+                color: c,
+                offset,
             };
         }).concat((this.tm.activeNode ? [{
             z: this.tm.activeNode.z,
-            color: [255, 255, 0, 200]
-        }]: [])));
+            color: [255, 255, 0, 200],
```
Comment (cabishop) [inline breadcrumbs/src/layers/Scrollbar.js]: Thoughts on making this color value a constant?

File: breadcrumbs/src/layers/Scrollbar.js [hunk 3]
```diff
@@ -71,15 +90,15 @@ export default class Scrollbar {
...
+                this.width + this.left + e.offset,
                 _z
             );
         });
 
         // Draw thumb:
         this.p.stroke(200);
         this.p.line(
-            this.left, this.top + this.height * (this.im.currentZ / this.im.nSlices),
+            this.left - 5, this.top + this.height * (this.im.currentZ / this.im.nSlices),
```
Comment (cabishop) [inline breadcrumbs/src/layers/Scrollbar.js]: @j6k4m8 will the offset here of "5" be the same as the one above? can you use the const HIGHLIGHT_OFFSET_AMOUNT ?

File: pointfog/src/layers/Scrollbar.js [hunk 4]
```diff
@@ -60,15 +70,15 @@ export default class Scrollbar {
...
+                this.width + this.left + e.offset,
                 _z
             );
         });
 
         // Draw thumb:
         this.p.stroke(200);
         this.p.line(
-            this.left, this.top + this.height * (this.im.currentZ / this.im.nSlices),
+            this.left - 5, this.top + this.height * (this.im.currentZ / this.im.nSlices),
```
Comment (cabishop) [inline pointfog/src/layers/Scrollbar.js]: Same comment as above in breadcrumbs/../../Scrollbar.js

File: pointfog/src/layers/Scrollbar.js [hunk 5]
```diff
@@ -33,14 +33,24 @@ export default class Scrollbar {
...
+            if (this.p.mouseX < this.left + this.width && this.p.mouseY < this.top + this.height) {
+                this.im.setZ(Math.round(this.im.nSlices * (this.p.mouseY - this.top) / (this.height - this.top)));
+                return false;
+            }
+        }
+        return true;
+    }
 
     getEntities() {
+        let offset = 5;
```
Comment (cabishop) [inline pointfog/src/layers/Scrollbar.js]: In breadcrumbs/../../Scrollbar.js you set the offset as a constant - fine either way, may be good to keep as much consistency in implementations where possible

File: pointfog/src/layers/Scrollbar.js [hunk 6]
```diff
@@ -33,14 +33,24 @@ export default class Scrollbar {
...
         return this.pm.nodes.map(i => {
             return {
                 z: i.z,
-                color: [0, 150, 200, 40]
+                color: i.bookmarked ? [250, 0, 150, 100] : [0, 150, 200, 40],
```
Comment (cabishop) [inline pointfog/src/layers/Scrollbar.js]: Consider using constants here for colors (e.g. BOOKMARKED_COLOR) if possible

File: breadcrumbs/src/layers/Scrollbar.js (hunk 3 above)
Comment (j6k4m8) [reply breadcrumbs/src/layers/Scrollbar.js]: Not the same — these are addressing different positioning artifacts. But also fine to make this a constant if you'd prefer it. It's only used once, though — and in a tight render loop.

Comment (cabishop) [reply breadcrumbs/src/layers/Scrollbar.js]: fine to leave as is then @j6k4m8

**Pull request merged**